src/main/java/net/snowflake/client/core/HttpClientSettingsKey.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2012-2021 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.client.core;
+
+import java.io.Serializable;
+import org.apache.http.HttpHost;
+
+/**
+ * This class defines all non-static parameters needed to create an HttpClient object. It is used as
+ * the key for the static hashmap of reusable http clients.
+ */
+public class HttpClientSettingsKey implements Serializable {
+
+  private OCSPMode ocspMode;
+  private boolean useProxy;
+  private String proxyHost = "";
+  private int proxyPort = 0;
+  private String nonProxyHosts = "";
+  private String proxyUser = "";
+  private String proxyPassword = "";
+
+  public HttpClientSettingsKey(
+      OCSPMode mode, String host, int port, String nonProxyHosts, String user, String password) {
+    this.useProxy = true;
+    this.ocspMode = mode != null ? mode : OCSPMode.FAIL_OPEN;
+    this.proxyHost = host.trim();
+    this.proxyPort = port;
+    this.nonProxyHosts = nonProxyHosts.trim();
+    this.proxyUser = user.trim();
+    this.proxyPassword = password.trim();
+  }
+
+  public HttpClientSettingsKey(OCSPMode mode) {
+    this.useProxy = false;
+    this.ocspMode = mode != null ? mode : OCSPMode.FAIL_OPEN;
+  }
+
+  @Override
+  public boolean equals(final Object O) {
+    if (O instanceof HttpClientSettingsKey) {
+      HttpClientSettingsKey comparisonKey = (HttpClientSettingsKey) O;
+      if (comparisonKey.ocspMode.getValue() == this.ocspMode.getValue()) {
+        if (!comparisonKey.useProxy) {
```

Comment:
Do we need to make sure this.useProxy is False as well?

Reply:
Good point- I added that in, and then added a sanity check test case called `CustomProxyLatestIT.testSizeOfHttpClientNoProxies` to ensure the hashtable isn't ballooning in size.